hbase-thrift/src/main/java/org/apache/hadoop/hbase/thrift/ThriftServer.java

```diff
@@ -249,6 +263,27 @@ protected void setupParamters() throws IOException {
     pauseMonitor.start();
   }
 
+  private String getSpengoPrincipal(Configuration conf, String host) throws IOException {
+    String principal = conf.get(THRIFT_SPNEGO_PRINCIPAL_KEY);
```

Comment:
We're doing this ourselves instead of using Hadoop Configuration's deprecation mechanisms because we want a different fall back order? Or we can't set deprecation soon enough for some reason?

We should have comments proactively letting future folks know the reasoning.

Reply:
Hmm let me take a look at what Hadoop Configuration's deprecation options are - I didn't know that was a fine. The old config is still technically valid just not for SPNEGO. I'll see if that is an option here to fallback with a message.

Reply:
So I don't think using the Hadoop `Configuration` deprecations will work here. The existing config key `hbase.thrift.kerberos.principal` is still valid and needs to be used for Kerberos communication between HBase Thrift Server to backend HBase master/rs. The new config key `hbase.thrift.spnego.principal` should be used for SPNEGO (handling SPNEGO HTTP only not for backend communication). 

So its not really a deprecation per say, but more configuration to split out the principal/keytab used for SPNEGO versus backend communication. In an ideal world, the config `hbase.thrift.kerberos.principal` would never have done double duty.

Reply:
I added a comment specifically for this case.

Reply:
Given I missed this the first time, the comment clearly explains things to me. Resolving.